jsdom-monkeypatch.js

```diff
@@ -0,0 +1,308 @@
+/**
+ * Reffy runs ReSpec in JSDOM to avoid having to download an run a complete
+ * headless browser.
+ *
+ * Problem is JSDOM does not yet support a number of features that ReSpec needs.
+ * Also, JSDOM does not allow interception of network requests anymore.
+ *
+ * This code monkey-patches JSDOM (and ReSpec) code before other modules make
+ * use of them, so that ReSpec can run in JSDOM and so that network requests
+ * use our fetch implementation.
+ *
+ * This module needs to be required before any other module that make use of
+ * JSDOM.
+ *
+ * The whole thing is PRETTY UGLY.
+ *
+ * NB: This code WILL LIKELY BREAK when switching to new versions of JSDOM or
+ * ReSpec.
```

Comment:
is there a simple test we can build to detect that early?

Reply:
For JSDOM, we control the switch to a new version, so running a crawl locally before pushing the version bump could be good-enough (but it would certainly be better to write a few test cases!).

For ReSpec, without test cases, it's a bit more difficult, because the latest version of ReSpec is used "on the fly". We could force the use of a specific version of ReSpec to make sure updates are explicit and remain under our control.